website/content/docs/concepts/architecture.mdx

```diff
@@ -162,14 +176,129 @@ on hardware features such as architecture and availability of GPUs, or software
+where the rules don't apply.
+
+Use affinities and constraints when certain jobs have certain node preferences
+but is acceptable to have other jobs sharing the same nodes.
+
+The sections below describe the node values that can be configured and used in
+job affinity and constraint rules.
+
+#### Node Class
+
+Node class is an arbitrary value that can be used to group nodes based on some
+characteristics, like the instance size or the presence of fast hard drives,
+and is specified in the client configuration file using the
+[`node_class`][config_client_node_class] parameter.
+
+#### Dynamic and Static Node Metadata
+
+Node metadata are arbitrary key-value mappings specified either in the client
+configuration file using the [`meta`][config_client_meta] parameter or
+dynamically via the [`nomad node meta`][cli_node_meta] command and the
+[`/v1/client/metadata`][api_client_metadata] API endpoint.
+
+There are no preconceived use cases for metadata values, and each team may
+choose to use them in different ways. Some examples of static metadata include
+resource ownership, such as `owner = "team-qa"`, or fine-grained locality,
+`rack = "3"`. Dynamic metadata may be used to track runtime information, such
+as jobs running in a given client.
+
+### Datacenter
+
+Datacenters represent a geographical location in a region that can be used for
+fault tolerance and infrastructure isolation.
+
+It is defined in the agent configuration file using the
+[`datacenter`][config_datacenter] parameter and, unlike affinities and
+constraints, datacenters are opt-in at the job level, meaning that a job only
+places allocations in the datacenters it uses, and, more importantly, only jobs
+in a given datacenter are allowed to place allocations in those nodes.
+
+Given the strong connotation of a geographical location, use datacenters to
+represent where a node resides rather than its intended use. The
+[`spread`][] block can help achieve fault tolerance across datacenters.
+
+### Node Pool
+
+Node pools allow grouping nodes that can be targeted by jobs to achieve
+workload isolation.
+
+Similarly to datacenters, node pools are configured in an agent configuration
+file using the [`node_pool`][config_client_node_pool] attribute, and are opt-in
+on jobs, allowing restricted use of certain nodes by specific jobs without
+extra configuration.
+
+But unlike datacenters, node pools don't have a preconceived notion and can be
+used for several use cases, such as segmenting infrastructure per environment
+(development, staging, production), by department (engineering, finance,
+support), or by functionality (databases, ingress proxy, applications).
+
+Node pools are also a first-class concept and can hold additional [metadata and
+configuration][spec_node_pool].
+
+Use node pools when there is a need to restrict and reserve certain nodes for
+specific workloads, or when you need to adjust specific [scheduler
+configuration][spec_node_pool_sched_config] values.
```

Comment:
Probably a nice place to plug that node pools can be associated with namespaces in Nomad Enterprise.

Reply:
Good idea. Added a blurb about it after moving this content.
https://github.com/hashicorp/nomad/blob/4f4ba9fdfc71eab2665c6091b249b711c6eb1f81/website/content/docs/concepts/scheduling/placement.mdx#L104-L105